### PR TITLE
update to Kubernetes 1.12.2

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/4709.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 
 [[constraint]]
-  branch = "master"
+  branch = "kubernetes-1.12.2"
```
Comment (kopiczko) [inline Gopkg.toml]: Revert.

**Comment by xh3b4sd** [comment]
Please check e2e.

**Comment by kopiczko** [comment]
Interestingly the caller in CI is `"caller":"github.com/giantswarm/e2e-harness/pkg/compiler/compiler.go:70"` but on my local machine it's `e2e-harness/pkg/compiler/compiler.go:71`. Do we have different versions?

**Comment by kopiczko** [comment]
OK `strings.Builder` was introduced in 1.10 https://golang.org/doc/go1.10#strings. Looks like we use `go` binary from the CI in `e2e-harness`.

File: .circleci/config.yml [hunk 2]
```diff
@@ -2,17 +2,15 @@ e2eTest: &e2eTest
...
-  - run: |
-      wget -q $(curl -sS https://api.github.com/repos/giantswarm/e2e-harness/releases/latest | grep browser_download_url | head -n 1 | cut -d '"' -f 4)
-      chmod +x ./e2e-harness
-  - run: ./e2e-harness localkube
-  - run: ./e2e-harness setup --remote=false
-  - run: ./e2e-harness test --test-dir=${TEST_DIR}
-  - run:
-      command: ./e2e-harness teardown
-      when: always
+    - checkout
```
Comment (tuommaki) [inline .circleci/config.yml]: Is there a specific reason why indentation changed here?

**Comment by xh3b4sd** [comment]
Which go version are we using on CI then? Can we update it?

File: .circleci/config.yml (hunk 2 above)
Comment (kopiczko) [reply .circleci/config.yml]: This is a leftover after me trying things. I will revert it. But this is the indentation circle recommends so I in general use it.

**Comment by kopiczko** [comment]
Some debugging info:

```
#!/bin/bash -eo pipefail
go version
go version go1.9.1 linux/amd64
```

```
#!/bin/bash -eo pipefail
which go
/usr/local/go/bin/go
```

```
#!/bin/bash -eo pipefail
echo $PATH
/opt/circleci/.nvm/versions/node/v6.1.0/bin:/opt/circleci/.rvm/gems/ruby-2.3.1/bin:/opt/circleci/.rvm/gems/ruby-2.3.1@global/bin:/opt/circleci/.rvm/rubies/ruby-2.3.1/bin:/home/circleci/.go_workspace/bin:/usr/local/go/bin:/opt/circleci/.pyenv/shims:/opt/circleci/.pyenv/bin:/opt/google-cloud-sdk/bin:/usr/local/apache-maven/bin:/home/circleci/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/local/gradle-1.10/bin:/opt/circleci/.rvm/bin:/opt/circleci/.rvm/bin
```

**Comment by kopiczko** [comment]
Works after merging https://github.com/giantswarm/e2e-harness/pull/167.

**Comment by kopiczko** [comment]
Closing in favour of https://github.com/giantswarm/operatorkit/pull/258.

**Pull request closed without merging**